### ISSUE TITLE
Look at integrating pagination in the MAAP API to improve UX for jobs ui

### ISSUE THREAD

**Issue by marjo-luc** [MAAP-Project/ZenHub#1027]
Loading jobs in the jobs view is very slow. Might want to also look into caching results for jobs that have completed as their status will likely not change.

**Comment by rtapella** [comment]
Along with this would be aggregation to help with filtering and reducing the listing... this is starting to feel like a more full-featured jobs UI and I wonder if we should reconsider where/how to implement the feature set (e.g. a standalone app, in HySDS, etc.)

**Comment by marjo-luc** [comment]
Yes -- we keep grappling with this. The jobs ui is meant to be a quick and easy way for users to gain some insight into their processing without needing to write code. Power users can always use Figaro, but that tool is not built into the user's workspace and there is a bit of a learning curve (not to mention you see what everyone's processing). 

The loose rule we've been going by is stopping short of implementing faceted search as that gets into Figaro territory (and I do not believe the MAAP API supports this -- might be mistaken, though). I think pagination is doable and certainly needs to be implemented in maap-py.

**Comment by marjo-luc** [comment]
Implemented in maap-py. [PR](https://github.com/MAAP-Project/maap-py/pull/96)
